tests/test_frame.py

```diff
@@ -2390,3 +2390,11 @@ def get_NDArray(df: pd.DataFrame, key: npt.NDArray[np.uint64]) -> pd.DataFrame:
 
     a: npt.NDArray[np.uint64] = np.array([10, 30], dtype="uint64")
     check(assert_type(get_NDArray(df, a), pd.DataFrame), pd.DataFrame)
+
+
+def test_change_to_dict_return_type() -> None:
+    id = [1, 2, 3]
+    value = ["a", "b", "c"]
+    df = pd.DataFrame(zip(id, value), columns=["id", "value"])
+    fd = df.set_index("id")["value"].to_dict()
+    check(assert_type(fd, dict[Any, Any]), dict)
```

Comment:
Need to capitalize `dict`

Reply:
Mypy still fails it shows expected type to be `Dict[Hashable, Any]`

Reply:
Should it be `dict[Any, S1]` because `flake8` complains about it `Y022 Use "dict[KeyType, ValueType]" instead of "typing.Dict[KeyType, ValueType]" (PEP 585 syntax)`

Reply:
> For `Series`, the type stub returns `dict[Any, S1]` , but for the test above, the type of the values of the series is unknown, so you should use `Dict[Any, Any]` in the `assert_type()`.
> 

oh, I got your point thanks